### ISSUE TITLE
[edn] parse error with escaped quotes in string

### ISSUE THREAD

**Issue by bgrins** [mozilla/mentat#434]
Found when attempting to import music data:

```
conn.transact(&mut sqlite, r#"[
        {:db/id 17592186049155, :artist/name "\"Quoted\""}
]"#).unwrap();
```

Leads to:

```
---- test_big stdout ----
	thread 'test_big' panicked at 'called `Result::unwrap()` on an `Err` value: Error(EdnParseError(ParseError { line: 2, column: 55, offset: 56, expected: {"[a-z]", "0", "\"", "/", "[A-Z]", "[12]", "\n", "(", ",", "[3]", "\t", "...", " ", "true", "[2-9]", "[0-9]", "0x", "#f", "\r", "[", "#{", "{", "[*!_?$%&=<>]", ";", "+", "[-*!_?$%&=<>]", "false", "nil", ".", ":", "-"} }), State { next_error: None, backtrace: None })', src/libcore/result.rs:859
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by ncalexan** [comment]
This is also a known issue: there's a comment about quoting in the code, but I don't see a ticket.

**Comment by bgrins** [comment]
OK, we can use this as the ticket for that work then

**Comment by rnewman** [comment]
Fixed in #589.
